### PR TITLE
make tests of pytorch_example device agnostic

### PR DESCRIPTION
**Pull request by ji-huazhong**
# What does this PR do?

Part of https://github.com/huggingface/transformers/issues/25654

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

cc @ydshieh @fxmarty @zhangsibo1129 @arsalanu

### CODE REVIEW

File: examples/pytorch/test_accelerate_examples.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def test_run_glue_no_trainer(self):
             --with_tracking
         """.split()
 
-        if is_cuda_and_apex_available():
+        if is_torch_fp16_available_on_device(torch_device):
             testargs.append("--fp16")
```
Comment (ydshieh) [inline examples/pytorch/test_accelerate_examples.py:97]: I don't know the reason why we test with `fp16` only if  `apex` is available. The change here will means we will test it as long as CI is running on GPU. Not sure if this is what we expect.

@muellerzr Do you happen to know the story here (about the `apex` part)?

Comment (muellerzr) [reply examples/pytorch/test_accelerate_examples.py:97]: I think it was a bad copy/paste, because I wouldn't have had that in there on purpose.

Comment (ydshieh) [reply examples/pytorch/test_accelerate_examples.py:97]: OK thank you for the feedback ❤️

**Review by ydshieh — APPROVED** [review]
Thank you @statelesshz for this work!

**Review by ydshieh — APPROVED** [review]

**Comment by ydshieh** [comment]
@ArthurZucker This is a continuation of (the merged) #25870. There might be more such PRs in the future.

**Review by zhangsibo1129 — APPROVED** [review]

**Review by amyeroberts — APPROVED** [review]
LGTM - thanks for adding this and improving our test suite!

**Pull request merged**